### PR TITLE
First edition of a bank dashboard

### PR DESCRIPTION
**Pull request by ElemRoy**

### CODE REVIEW

**Review by npelenskii — COMMENTED** [review]
Check your eslint when you will resolve conflicts

**Review by npelenskii — APPROVED** [review]

**Review by ShevtsovVolodymyr — APPROVED** [review]

File: src/components/UIElements/BasketDashboardHeader.jsx [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+  "Cheap to expensive",
+  "Divider",
+  "Soon to expire",
+  "Far to expire"
+]
+
+const BasketDashboardHeader = () => {
+  const { page } = useSelector(state => state.basket.paginationData);
+
+  const [hiddenFilterAnchor, sethiddenFilterAnchor] = React.useState(null);
```
Comment (AntonDeXy) [inline src/components/UIElements/BasketDashboardHeader.jsx:58]: as you import useEffect directly from react, not React.useEffect, then do the same thing for use state

File: src/components/UIElements/BasketDashboardHeader.jsx [hunk 2]
```diff
@@ -0,0 +1,179 @@
+import React from 'react';
+import { useEffect } from 'react';
```
Comment (AntonDeXy) [inline src/components/UIElements/BasketDashboardHeader.jsx:2]:
```suggestion
import React, { useEffect }  from 'react';
```

File: src/pages/Basket.jsx [hunk 3]
```diff
@@ -27,7 +30,23 @@ const BorderLinearProgress = styled(LinearProgress)(({ theme }) => ({
...
 
+const getDaysBetweenDates = (date1, date2) => {
+  const differenceInTime = date1 - date2;
+  const differenceInDays = Math.floor(differenceInTime / (1000 * 3600 * 24));
+
+  return differenceInDays;
+}
+
 const Basket = () => {
+  const id = useParams();
```
Comment (AntonDeXy) [inline src/pages/Basket.jsx:41]: useParams() returns object, not a value, am I right?

File: src/pages/Basket.jsx [hunk 4]
```diff
@@ -27,7 +30,23 @@ const BorderLinearProgress = styled(LinearProgress)(({ theme }) => ({
...
+
+  const { name, ownerId: { firstName }, description, goal, value, expirationDate, isPublic, createdAt, image } = useSelector(state => state.basket.basket);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(get_basket_by_id({ id: id.basket }))
```
Comment (AntonDeXy) [inline src/pages/Basket.jsx:47]: what is in id? Can you pls rename it, cause it looks so weird when u taking basket from id

File: src/redux/basket/basketActions.js [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+        { params: { archived, page, order } },
+        );
+
+        if(!req.data.basketData){
+          return rejectWithValue({ message: "There is an error with getting baskets" });
+        }
+  
+        return {basketData: req.data.basketData, paginationData: req.data.paginationData};
+    } catch (error) {
+        if (error.response && error.response.data.message) {
```
Comment (AntonDeXy) [inline src/redux/basket/basketActions.js:80]: As for me it will be better to use `error?.response.data.message `

Comment (AntonDeXy) [reply src/redux/basket/basketActions.js:80]: check other places too

File: src/services/axios/index.js [hunk 6]
```diff
@@ -6,7 +6,8 @@ import { setError } from '../../redux/request/requestSlice';
 const errorsBlacklist = ['/api/auth/refresh'];
 
 const instance = axios.create({
-  baseURL: process.env.REACT_APP_API_URL,
+  //baseURL: process.env.REACT_APP_API_URL,
```
Comment (AntonDeXy) [inline src/services/axios/index.js:9]: why did you hardcoded this value?

File: src/components/UIElements/BasketDashboardHeader.jsx [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+  "Cheap to expensive",
+  "Divider",
+  "Soon to expire",
+  "Far to expire"
+]
+
+const BasketDashboardHeader = () => {
+  const { page } = useSelector(state => state.basket.paginationData);
+
+  const [hiddenFilterAnchor, sethiddenFilterAnchor] = React.useState(null);
```
Comment (AntonDeXy) [inline src/components/UIElements/BasketDashboardHeader.jsx:57]: ?

File: src/components/UIElements/BasketDashboardHeader.jsx [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+
+  const handleFilterClick = (event) => {
+    sethiddenFilterAnchor(event.currentTarget);
+  };
+  const handleFilterClose = () => {
+    sethiddenFilterAnchor(null);
+  };
+
+  const { currentType, currentOrder }= useSelector(state => state.basket.paginationData);
+  const [archivedBasketsState, setarchivedBasketsState] = React.useState(false);
```
Comment (AntonDeXy) [inline src/components/UIElements/BasketDashboardHeader.jsx:68]: ?

File: src/pages/Basket.jsx (hunk 3 above)
Comment (AntonDeXy) [reply src/pages/Basket.jsx]: ?

Comment (AntonDeXy) [reply src/pages/Basket.jsx]: if yes, then maybe it can be named in a different way?

File: src/pages/Basket.jsx (hunk 4 above)
Comment (AntonDeXy) [reply src/pages/Basket.jsx]: ?

File: src/components/UIElements/BasketDashboard.jsx [hunk 9]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import BasketBox from './BasketBox';
+import { useEffect, useState } from 'react';
```
Comment (AntonDeXy) [inline src/components/UIElements/BasketDashboard.jsx:3]:
```suggestion
import React,  { useEffect, useState } from 'react';
import BasketBox from './BasketBox';
```

File: src/pages/Basket.jsx [hunk 10]
```diff
@@ -1,4 +1,5 @@
 import React from 'react';
+import { useEffect } from 'react';
```
Comment (AntonDeXy) [inline src/pages/Basket.jsx:2]: write in in one line

Comment (AntonDeXy) [reply src/pages/Basket.jsx:2]: as one import*

**Pull request merged**